src/component/mxgraph/GraphCellUpdater.ts

```diff
@@ -94,26 +95,50 @@ export default class GraphCellUpdater {
         let cellStyle = cell.getStyle();
         // Only set the style when the key is set. Otherwise, mxGraph removes the related setting from the cellStyle which is equivalent to a reset of the style property
         styleUpdate.stroke?.color && (cellStyle = mxgraph.mxUtils.setStyle(cellStyle, mxgraph.mxConstants.STYLE_STROKECOLOR, styleUpdate.stroke.color));
+        styleUpdate.font && (cellStyle = this.updateFont(styleUpdate.font, cellStyle));
 
-        const font = styleUpdate.font;
-        if (font) {
-          font.color && (cellStyle = mxgraph.mxUtils.setStyle(cellStyle, mxgraph.mxConstants.STYLE_FONTCOLOR, font.color));
-          font.size && (cellStyle = mxgraph.mxUtils.setStyle(cellStyle, mxgraph.mxConstants.STYLE_FONTSIZE, font.size));
-          font.family && (cellStyle = mxgraph.mxUtils.setStyle(cellStyle, mxgraph.mxConstants.STYLE_FONTFAMILY, font.family));
-
-          font.isBold !== undefined && (cellStyle = mxgraph.mxUtils.setStyleFlag(cellStyle, mxgraph.mxConstants.STYLE_FONTSTYLE, mxgraph.mxConstants.FONT_BOLD, font.isBold));
-          font.isItalic !== undefined && (cellStyle = mxgraph.mxUtils.setStyleFlag(cellStyle, mxgraph.mxConstants.STYLE_FONTSTYLE, mxgraph.mxConstants.FONT_ITALIC, font.isItalic));
-          font.isUnderline !== undefined &&
-            (cellStyle = mxgraph.mxUtils.setStyleFlag(cellStyle, mxgraph.mxConstants.STYLE_FONTSTYLE, mxgraph.mxConstants.FONT_UNDERLINE, font.isUnderline));
-          font.isStrikeThrough !== undefined &&
-            (cellStyle = mxgraph.mxUtils.setStyleFlag(cellStyle, mxgraph.mxConstants.STYLE_FONTSTYLE, mxgraph.mxConstants.FONT_STRIKETHROUGH, font.isStrikeThrough));
-
-          // TODO To uncomment when we implement the Opacity in global/background/font/stroke
-          // font.opacity && (cellStyle = mxgraph.mxUtils.setStyle(cellStyle, mxgraph.mxConstants.STYLE_TEXT_OPACITY, font.opacity));
+        if (isShapeStyleUpdate(styleUpdate)) {
+          styleUpdate.fill && (cellStyle = this.updateFill(styleUpdate.fill, cellStyle));
         }
 
         this.graph.model.setStyle(cell, cellStyle);
       }
     });
   }
+
+  private updateFont(font: Font, cellStyle: string): string {
```

Comment:
**suggestion:** I suggest we move the method out of the class, we rename the function and we reorder the parameters. The same applies to the `updateFill` method.

The method doesn't access to any properties of the class, so it can be move outside of the class.
I didn't understand at first glance that it modifies all font properties. So I suggest to rename it into `updateFontProperties` or `updateFontStyleProperties`.
I also suggest to put the cellStyle parameter in the first position. This is the element which is updated.
With this change the reading the code better reflects what is done: "update the font" (function name) of the "cellStyle" (the 1st parameter) with the provided "font" info (the 2nd parameter).

Reply:
Decision: change the parameter order and move the method out of the class.
No need for renaming after changing the parameters order.